eth/fetcher/tx_fetcher_test.go

```diff
@@ -304,7 +304,7 @@ func TestTransactionFetcherSingletonRequesting(t *testing.T) {
 func TestTransactionFetcherFailedRescheduling(t *testing.T) {
 	// Create a channel to control when tx requests can fail
 	proceed := make(chan struct{})
-
+	defer close(proceed)
```

Comment:
IMHO this is a bad change. The `proceed` channel is used to tightly control when the fetchers can tick ahead. If there's a leftover hanging, waiting to proceed, we should figure out if it's correct or not. And if correct, we should add a doFunc step that permits it to continue. If it's incorrect, we need to find why. Closing the channel just blindly unblocks stuff that might be incorrectly blocked.